lnrpc/routerrpc/router_server.go

```diff
@@ -178,22 +185,26 @@ func (s *Server) RegisterWithRootServer(grpcServer *grpc.Server) error {
 // PaymentRequest, then an error will be returned. Otherwise, the payment
 // pre-image, along with the final route will be returned.
 func (s *Server) SendPayment(ctx context.Context,
-	req *PaymentRequest) (*PaymentResponse, error) {
+	req *SendPaymentRequest) (*SendPaymentResponse, error) {
 
 	payment, err := s.cfg.RouterBackend.extractIntentFromSendRequest(req)
 	if err != nil {
 		return nil, err
 	}
 
-	preImage, _, err := s.cfg.Router.SendPayment(payment)
-	if err != nil {
-		return nil, err
-	}
+	// Launch the payment in a goroutine. It is no problem that we discard
+	// the error, because errors will also be recorded by control tower and
+	// available through LookupPayment. Exception to this are internal
+	// errors, but those should not happen.
+	go func() {
+		_, _, err := s.cfg.Router.SendPayment(payment)
```

Comment:
I think we should rather make `SendPayment` return when it has registered the payment with the control tower. That way the caller know that after this RPC call returns without an error, the payment is being handled.

Reply:
This is also racy, since we have no guarantee this goroutine will run before we call `LookupPayment`.

Reply:
Yes you are totally right. I had it like you suggested, but rebased it out mistakingly thinking it was equivalent.